sys/net/application_layer/nanocoap/sock.c

```diff
@@ -102,41 +136,100 @@ ssize_t nanocoap_sock_request(nanocoap_sock_t *sock, coap_pkt_t *pkt, size_t len
                 continue;
             }
 
-            break;
+            switch (coap_get_type(pkt)) {
+            case COAP_TYPE_RST:
+                /* TODO: handle different? */
+                return -EBADMSG;
+            case COAP_TYPE_CON:
+                /* TODO: send ACK response */
```

Comment:
Without the ACK the communication partner will re-transmit the CON for roughly 5 minutes or so with default parameters. That really should be done to allow the partner to free the resources earlier and put less load on the network.

Reply:
Is sending an empty ACK with code 2.03 the right thing to do here?

Reply:
An empty ACK contains no code, as it is is only used when a meaningful reply is not yet possible.

I'm currently on my phone. If I recall correctly it is about a separate response. To that, one would react with an empty ACK (without code), as what was received is already the response (and there is no sensible reply to a response in the request-response model of CoAP). So the empty ACK in that case only is used to cease retransmissions.